crates/sncast/src/helpers/configuration.rs

```diff
@@ -84,10 +88,22 @@ pub struct CastConfig {
     pub networks: NetworksConfig,
 }
 
+impl CastConfig {
+    pub fn validate(&self) -> anyhow::Result<()> {
+        match (&self.url, &self.network) {
+            (Some(_), Some(_)) => {
+                anyhow::bail!("Only one of `url` or `network` may be specified")
+            }
+            _ => Ok(()),
+        }
```

Comment:
Maybe let's add an todo for your issue somewhere in code? https://github.com/foundry-rs/starknet-foundry/issues/4027